aws-provider/postgresdb/README.md

```diff
@@ -0,0 +1,346 @@
+
+     ```bash
+     aws iam list-roles | jq '.Roles[] | select( .RoleName == "crossplane-assume-role")'
+     ```
+
+     If the role does not exist, we need to create it and attach the policy
+
+     ```bash
+       aws iam create-role --role-name crossplane-assume-role --policy-document file://policies/crossplane-assume-role-trust-policy.json
+       aws iam attach-role-policy --policy-arn \
+           "arn:aws:iam::aws:policy/crossplane-assume-role-policy" \
+           --role-name crossplane-assume-role
+     ```
+
+     If the role already exists, update its trust policy with
+
+     ```bash
+       aws iam update-assume-role-policy --role-name crossplane-assume-role \
+           --policy-document file://policies/crossplane-assume-role-trust-policy.json
+     ```
+
+## Create the secondary role
+
+Use the steps defined in [creating the primary role](#create-the-primary-role) to
+create a new role `rds-crossplane-role`
+
+This second role will use the attach the trust policy and IAM permissions policy
+
+- [rds-crossplane-policy](./policies/rds-crossplane-policy.json).
+- [rds-crossplane-role-trust-policy](./policies/rds-crossplane-role-trust-policy.json)
+
+> **Note** Creating IAM policies to be assumed by other roles
+>
+> When creating roles to be assumed by the service account identity role
+> `crossplane-assume-role`, at the very least, the policy must include the
+> following statement for other roles to be allowed to assume it.
+>
+> ```json
+> {
+>     "Sid": "VisualEditor2",
+>     "Effect": "Allow",
+>     "Action": "sts:AssumeRole",
+>     "Resource": "*",
+>     "Condition": {
+>         "ForAnyValue:StringLike": {
+>             "aws:PrincipalArn": [
+>                 "arn:aws:sts::${AWS_ACCOUNT_ID}:role/crossplane-assume-role",
+>                 "arn:aws:sts::${AWS_ACCOUNT_ID}:assumed-role/rds-crossplane-role*"
+>             ]
+>         }
+>     }
+> }
+> ```
+>
+> Without this, crossplane may not be able to assume the role and you will
+> experience permissions failures when building resources.
+
+The policy file [rds-crossplane-policy](./policies/rds-crossplane-policy.json)
+is a restrictive policy that allows only the specific actions required to build
+this example.
+
+## Enabling providers
+
+Next, we need to set up the providers.
+
+In your management clusters git repository, go to the `management-clusters/MC_NAME/crossplane-providers`
+folder and create `kustomization.yaml` file in a sub-directory named after the
+apigroup of your provider. For example:
+
+```nohighlight
+├── crossplane-providers
+│   ├── ec2
+│   │   └── kustomization.yaml
+│   ├── kms
+│   │   └── kustomization.yaml
+│   └── rds
+│       └── kustomization.yaml
+```
+
+into each `kustomization.yaml` file, copy the following contents:
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - https://github.com/giantswarm/management-cluster-bases//extras/crossplane/providers/upbound/aws?ref=main
```

Comment:
This gives me 404, should be `https://github.com/giantswarm/management-cluster-bases/tree/main/extras/crossplane/providers/upbound/aws?ref=main` ?

Reply:
No because this is a command to `kustomize` which is copying the remote refs (See [kustomize examples](https://github.com/kubernetes-sigs/kustomize/blob/master/examples/remoteBuild.md#examples))

* With `tree/main` included:
   ```
   $ kustomize build .
   Error: accumulating resources: accumulating resources from 
   'https://github.com/giantswarm/management-cluster-bases/tree/main/extras/crossplane/providers/upbound/aws?ref=main': 
   MalformedYAMLError: yaml: line 203: mapping values are not allowed in this context in File: 
   https://github.com/giantswarm/management-cluster-bases/tree/main/extras/crossplane/providers/upbound/aws?ref=main
   ```
   
 * Without:
    ```
   $ kustomize build .
   apiVersion: v1
   kind: ServiceAccount
   metadata:
     annotations:
   ...
    ```